### ISSUE TITLE
Exscript.protocols.Exception.TimeoutException: Buffer:

### ISSUE THREAD

**Issue by ebeng** [knipknap/exscript#149]
Hi,

I wrote a little script to run on Cisco devices, to retrieve show commands from it. But it is bumping into an error with the buffer and showing the Banner of the device. Script is as follows:

>         print ('Logging in')
>         account = Account(username, password)
>         session = SSH2(debug = 1)
>         session.connect(host)
>         session.login(account)
>         session.send("enable\r")
>         session.auto_app_authorize(account)
> 

This is what i am getting back:

> Password:
> Logging in
> generic: Rejecting ssh-rsa host key for xxxx: xxxx
> generic: Attempting to authenticate xxxx.
> generic: Authenticating with _paramiko_auth_password
> generic: Attempting to app-authenticate xxxx.
> generic: Expecting a prompt
> one_os: Protocol: driver replaced: generic -> one_os
> one_os: Protocol.app_authenticate(): driver replaced
> one_os: Expecting a prompt
> one_os: Shell prompt received.
> one_os: Expecting a prompt
> one_os: Calling driver.auto_authorize().
> one_os: Attempting to app-authorize xxxx.
> one_os: Expecting a prompt
> Traceback (most recent call last):
>   File "./BRASIL_cmd_saver", line 109, in <module>
>     main()
>   File "./BRASIL_cmd_saver", line 99, in main
>     response = get_commando(switch, username, password)
>   File "./BRASIL_cmd_saver", line 36, in get_commando
>     session.login(account)
>   File "/usr/lib/python2.6/site-packages/Exscript-DEVELOPMENT-py2.6.egg/Exscript/protocols/Protocol.py", line 609, in login
>     self.auto_app_authorize(app_account, flush = flush)
>   File "/usr/lib/python2.6/site-packages/Exscript-DEVELOPMENT-py2.6.egg/Exscript/protocols/Protocol.py", line 858, in auto_app_authorize
>     self.get_driver().auto_authorize(self, account, flush, bailout)
>   File "/usr/lib/python2.6/site-packages/Exscript-DEVELOPMENT-py2.6.egg/Exscript/protocols/drivers/one_os.py", line 43, in auto_authorize
>     conn.app_authorize(account, flush, bailout)
>   File "/usr/lib/python2.6/site-packages/Exscript-DEVELOPMENT-py2.6.egg/Exscript/protocols/Protocol.py", line 832, in app_authorize
>     self._app_authenticate(account, password, flush, bailout)
>   File "/usr/lib/python2.6/site-packages/Exscript-DEVELOPMENT-py2.6.egg/Exscript/protocols/Protocol.py", line 702, in _app_authenticate
>     raise TimeoutException(msg)
> Exscript.protocols.Exception.TimeoutException: Buffer: '\r\n   Device name :   xxxxxxxxx \r\n\r\nSWITCH01>'

**Comment by knipknap** [comment]
The reason is probably that you are sending the "enable" command, after which the device sends a "Password:" prompt.

http://exscript.readthedocs.io/en/latest/Exscript.protocols.html#Exscript.protocols.Protocol.auto_app_authorize

`auto_app_authorize` does the following steps:
1. it waits for a standard prompt
2. upon getting the prompt, it sends "enable\r"
3. it then waits for "Password"
4. Then it sends the password

Since you already sent "enable\r", the first step fails.

In other words, remove this line from your script, and it should work:

`session.send("enable\r")`

**Comment by ebeng** [comment]
It is still failing the same way. i just tried on some different switches and routers.

**Comment by knipknap** [comment]
Could you show the log after removing `connection.send('enable')`, and with debug=5

**Comment by ebeng** [comment]
https://paste.ee/p/tTjNQ

**Comment by knipknap** [comment]
Alright, the reason is that your device is incorrectly recognized as a router running OneOS, when it should be recognized as IOS. The reason for that is that your host appears to be using a (for Cisco IOS) unusual login prompt that **can't** be distinguished from OneOS.

There are three ways around that:
a) Force the OS to IOS after logging in: `conn.set_driver('ios')` does that.
b) Configure a banner on the Cisco host that includes the string "User Access Verification"
c) Disable the OneOS driver before connecting: `from Exscript.protocols.drivers import disable_driver; disable_driver('one_os')`. This requires the latest Exscript from master, it's not yet in a release.

If your router is actually running OneOS and not IOS, I can patch Exscript to fix this, but please provide

a) a complete login log with debug=0 and
b) the output of "show version"

in that case.

**Comment by ebeng** [comment]
sorry to come back so late back again to this topic.

I think the problem exist in that after the login, we have another "banner exec", where we give the user, more information about the device. 

So first we have a Disclaimer "banner login" where you say, its not prohibited to login blabla. After the successfull login there is another banner, with site information. 

I see that other scripts are going oke when there is no "banner exec" configured on the Cisco.
